sgx/TESTING.md

```diff
@@ -96,3 +94,27 @@ then the server does the same.
 The key sharing messages are in a custom format containing:
 * the ephemeral public key point in octet string format
 * a signature digest for the octet string, signed by the persistent private key
+
+
+### ECDHE/TLS Proxy Application
+
+In addition to the ECDHE test key server,
+Kmyth also includes a proxy application supporting connections
+from the SGX demo program to any remote TLS service.
+
+This proxy is intended for test purposes only.
+It functions as a man-in-the-middle,
+so all message payloads are visible to the proxy,
+which does not use trusted hardware to protect data in use.
+The proxy should only be used for testing and demonstration purposes.
+
+The ECDHE key exchange format is identical to that used in the ECDHE test key server,
+which is described above.
+
+#### Usage
+
+The proxy usage is similar to the test key server application.
+
+```
+./demo/bin/tls-proxy -r data/server_priv_test.pem -u data/client_cert_test.pem -p LOCAL_PORT -I REMOTE_HOST -P REMOTE_PORT -C REMOTE_CA_CERT -R CLIENT_KEY -U CLIENT_CERT
```

Comment:
```data/server_priv_test.pem``` and ```data/client_cert_test.pem``` should probably be SERVER_KEY and CLIENT_CERT  

Additionally, it's not clear to me what REMOTE, SERVER, and CLIENT would be in this context. Does ```-R CLIENT_KEY``` mean that the proxy server needs every client's private key?

Reply:
Here's one way to make those names more clear and consistent.

```
./demo/bin/tls-proxy -r ECDH_LOCAL_KEY -u ECDH_REMOTE_CERT -p ECDH_LOCAL_PORT -I TLS_REMOTE_HOST -P TLS_REMOTE_PORT -C TLS_REMOTE_CA_CERT [ -R TLS_LOCAL_KEY -U TLS_LOCAL_CERT ]
```

The `-R` argument is an RSA private key in PEM format, and the `-U` argument is its matching RSA public key PEM cert.
Both are required for authentication with PyKMIP, but they may not be required for other TLS services.

The `-U` cert must be signed by a CA, and the PyKMIP server must have a copy of the CA cert (the `ca_path` field in the PyKMIP server config).
It should be possible to use the same local key for both ECDH and TLS authentication, but I think it's better to keep them separate.